src/kernels/kernelFinder.base.ts

```diff
@@ -51,22 +52,17 @@ export abstract class BaseKernelFinder implements IKernelFinder {
     public async rankKernels(
         resource: Resource,
         notebookMetadata?: nbformat.INotebookMetadata,
+        preferredInterpreter?: PythonEnvironment,
         cancelToken?: CancellationToken,
-        useCache?: 'useCache' | 'ignoreCache'
+        useCache?: 'useCache' | 'ignoreCache',
+        serverId?: string
```

Comment:
Added a new argument `serverId` so that we can filter out the kernels that apply to a specific remote jupyter server.
Applies to the lazy controller that spins up jupyter and then asks us to get the prefered controller (surely at that point we need to return the preferred controller specific to that jupyter server)